osu.Game/Screens/SelectV2/BeatmapCarouselFilterGrouping.cs

```diff
@@ -225,18 +233,52 @@ private List<GroupMapping> getGroups(List<CarouselItem> items, FilterCriteria cr
                     return getGroupsBy(b => defineGroupBySource(b.BeatmapSet!.Metadata.Source), items);
 
                 case GroupMode.Collections:
-                    var collections = getCollections?.Invoke() ?? Enumerable.Empty<BeatmapCollection>();
-                    return getGroupsBy(b => defineGroupByCollection(b, collections), items);
+                {
+                    var realm = getRealm?.Invoke();
+
+                    return realm?.Run(r =>
+                    {
+                        var collections = r.All<BeatmapCollection>().AsEnumerable();
+                        return getGroupsBy(b => defineGroupByCollection(b, collections), items);
+                    }) ?? new List<GroupMapping>();
+                }
 
                 case GroupMode.MyMaps:
                     return getGroupsBy(b => defineGroupByOwnMaps(b, criteria.LocalUserId, criteria.LocalUserUsername), items);
 
+                case GroupMode.RankAchieved:
+                {
+                    var realm = getRealm?.Invoke();
+
+                    var topRankMapping = new Dictionary<BeatmapInfo, ScoreRank>(items.Count);
+
+                    return realm?.Run(r =>
+                    {
+                        var allLocalScores = r.All<ScoreInfo>()
+                                              .Filter($"{nameof(ScoreInfo.User)}.{nameof(RealmUser.OnlineID)} == $0"
+                                                      + $" && {nameof(ScoreInfo.BeatmapInfo)}.{nameof(BeatmapInfo.Hash)} == {nameof(ScoreInfo.BeatmapHash)}"
```

Comment:
What's this conditional?

Reply:
Carried across from `TopLocalRank` (also exists in `LeaderboardManager`), it seems to be for discarding scores of a beatmap version that doesn't match the current one.

Reply:
Inline comment mentioning this is the case, please. I can't read brains.